test/integration/MExtensionSystem.t.sol

```diff
@@ -0,0 +1,1548 @@
+                DEFAULT_ADMIN_ROLE
+            )
+        );
+        mEarnerManager.grantRole(EARNER_MANAGER_ROLE, multiRoleUser);
+
+        yieldAfter = mEarnerManager.accruedYieldOf(alice);
+        assertTrue(yieldAfter > 0, "Should have accrued yield");
+    }
+
+    function _calcMEarnerManagerPrincipal(uint256 amount) public view returns (uint112) {
+        uint128 _index = _currentMIndex();
+
+        return IndexingMath.getPrincipalAmountRoundedDown(uint240(amount), _index);
+    }
+
+    function _calcMEarnerManagerYield(uint256 balance, uint112 principal) public view returns (uint256) {
+        uint128 currentIndex = _currentMIndex();
+
+        uint256 balanceWithYield = IndexingMath.getPresentAmountRoundedUp(principal, currentIndex);
+
+        // Yield is the difference between present value and current balance
+        return balanceWithYield > balance ? balanceWithYield - balance : 0;
+    }
+
+    function _calcMYieldFeePrincipal(uint256 amount) public view returns (uint112) {
+        uint128 _index = _currentMYieldFeeIndex();
+
+        return IndexingMath.getPrincipalAmountRoundedUp(uint240(amount), _index);
+    }
+
+    function _calcMYieldFeeYield(uint256 priorAmount, uint112 _principal) public view returns (uint256) {
+        uint128 _index = _currentMYieldFeeIndex();
+
+        uint256 _amountPlusYield = IndexingMath.getPresentAmountRoundedUp(_principal, _index);
+
+        return _amountPlusYield - priorAmount;
+    }
+
+    function _currentMYieldFeeIndex() public view returns (uint128) {
+        unchecked {
+            return
+                // NOTE: Cap the index to `type(uint128).max` to prevent overflow in present value math.
+                UIntMath.bound128(
+                    ContinuousIndexingMath.multiplyIndicesDown(
+                        mYieldFeeIndexInitial,
+                        ContinuousIndexingMath.getContinuousIndex(
+                            ContinuousIndexingMath.convertFromBasisPoints(mYieldFeeRate),
+                            uint32(vm.getBlockTimestamp() - mYieldFeeIndexStart)
+                        )
+                    )
+                );
+        }
+    }
+
+    function divideUp(uint240 x, uint128 index) internal pure returns (uint112 z) {
+        unchecked {
+            // NOTE: While `uint256(x) * EXP_SCALED_ONE` can technically overflow, these divide/multiply functions are
+            //       only used for the purpose of principal/present amount calculations for continuous indexing, and
+            //       so for an `x` to be large enough to overflow this, it would have to be a possible result of
+            //       `multiplyDown` or `multiplyUp`, which would already satisfy
+            //       `uint256(x) * EXP_SCALED_ONE < type(uint240).max`.
+            return UIntMath.safe112(((uint256(x) * EXP_SCALED_ONE) + index - 1) / index);
+        }
+    }
+
+    function _calcMPrincipalAmountRoundedUp(uint256 amount) public view returns (uint112) {
```

Comment:
Similar to `_calcMYieldFeePrincipal()`.

Reply:
`_calcMYieldFeePrincipal()` uses `_currentMYieldFeeIndex();` instead of `_currentMIndex()`

Reply:
Ah ok. You could combine and just pass the correct index to the function.